torch/csrc/jit/codegen/onednn/guard_shape.cpp

```diff
@@ -0,0 +1,147 @@
+#include <torch/csrc/jit/codegen/onednn/guard_shape.h>
+
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/tensorexpr_fuser.h>
+#include <torch/csrc/jit/passes/utils/subgraph_utils.h>
+#include <torch/csrc/jit/runtime/graph_executor.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace onednn {
+
+using tensor_type_converter_t =
+    c10::function_ref<TensorTypePtr(const TensorTypePtr& t)>;
+
+void insertTypeGuardForFusionGroup(
```

Comment:
Could we just use https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/tensorexpr_fuser.cpp directly ? what is preventing that

Reply:
This code is identical to `insertTypeGuard`, except that it also skips guarding inputs that are outputs of `prim::oneDNNFusionGroup`. 

So, can we add an `if` condition to `insertTypeGuard` pertaining to `prim::oneDNNFusionGroup` [here](https://github.com/pytorch/pytorch/blob/72cedba6553d6fdacc55ccac0b1b70446c03ef82/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L284-L287)?

Reply:
Oh, thanks! I see your point. The input value's node would be `prim::If` and not `prim::FusionGroup` (or in this case `prim::oneDNNFusionGroup`), so no fusion group would directly output a value into another!

I'll reuse existing code, as you advised.